code/game/objects/effects/effect_system/smoke.dm

```diff
@@ -11,6 +11,7 @@
 	layer = FLY_LAYER
 	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
 	pass_flags = PASS_AIR
+	resistance_flags = UNACIDABLE|PLASMACUTTER_IMMUNE|PROJECTILE_IMMUNE|CRUSHER_IMMUNE
```

Comment:
?

Reply:
explosions simply ignore indestructable flagged atoms. I don't think there should be any other way of actually hurting smoke though.